packages/hash/api/src/db/adapter.ts

```diff
@@ -161,18 +161,17 @@ export type Graph = {
   }[];
 };
 
+export type DbTextProperties = {
+  /** @todo: find better type for this */
+  tokens: JSONObject[];
+};
```

Comment:
@nathggns is there any better way to type this at the moment? As far as I can tell, we're typing this as an arbitrary JSON object wherever it's used.

Reply:
This type exists in `shared/src/graphql/types.ts`. I don't remember why we specified this as a scalar in GraphQL though. @kachkaev led on this so he may recall.

Reply:
AFAIU, any feal type in GraphQL is a scalar, even if it is a structure. The resolvers pick up the right shape correctly, thanks to codegen.

Reply:
It could be to do with the discriminated union that we use inside `TextToken`. 

https://github.com/hashintel/hash/blob/40e1fd0204304f3fef905cf691114772a9130d6e/packages/hash/shared/src/graphql/types.ts#L1-L12

Graph queries for this structure would be quite bulky because we'd need to mention every field in every ‘case’ statement there. We don't need to filter fields inside resolvers for the text token, so it's kind of redundant (at least ATM).

Reply:
Thanks both, that was the information I was looking for :)

I've opted for directly importing `TextToken` from the shared package (rather than using the `TextProperties` GQL type) as in the past we've tried to decouple the db adapter types from the GQL generated types, as they serve different purposes and don't necessarily align.

Have made that change here: https://github.com/hashintel/hash/pull/303/commits/5fe713d00ae61aa5a02929757f42d5f54d3bdf5a